lib/base-compiler.ts

```diff
@@ -2225,7 +2231,9 @@ export class BaseCompiler implements ICompiler {
 
         if (!bypassCache) {
             const cacheRetreiveTimeStart = process.hrtime.bigint();
-            const result = await this.env.cacheGet(key);
+            // TODO: Because key coantains a CompilerInfo which contains a function member it can't be assigned to a
```

Comment:
what is the "todo" part? This seems like a reason to cast to any for sure, but I'm not sure what the action to take is.

Reply:
TODO to eliminate the ugly `any` cast, `key` should be cacheable (if it's not that's a big problem), it's just a matter of getting the types lined up

Reply:
I'll update it to be more clear